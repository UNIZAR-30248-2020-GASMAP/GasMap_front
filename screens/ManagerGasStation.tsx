import * as React from 'react';
import { Alert, Image, StyleSheet, ScrollView, Modal, Picker, TextInput } from 'react-native';

import EditScreenInfo from '../components/EditScreenInfo';
import { Text, View } from '../components/Themed';
import { Icon, Button } from 'react-native-elements';

import { getGasStationsById, updateGasServices, allServices, updatePriceMan } from '../drivers/connection';
import Table from 'react-native-simple-table';
import { LineChart, XAxis, YAxis, Grid } from 'react-native-svg-charts';
import { ListItem, Avatar } from 'react-native-elements';
import { Ionicons } from '@expo/vector-icons';
import { servicesListToIcon } from '../models/listServicesToIcon'
import { DeviceEventEmitter } from 'react-native'





export const FORMAT_BACK = "FORMAT_BACK";
export const FORMAT_EDIT = "FORMAT_EDIT";

export default class ManagerGasStation extends React.Component {


  constructor(props: any) {
    super(props)
    this.state = {
      datosGasolinera: [],
      allServices: [],
      horario: "",
      modalUpdateVisible: false,
      fuel: "Gas",
      price: "",
      fuelType: "" //for ghart
    };
  }

  // const willFocusSubscription: any = "";

  async componentDidMount() {
    console.log("COMPONENT DID MOUNT")
    this.getData();
    //   // DeviceEventEmitter.addListener('Listener data', async (e) => {
    //   //   //reload data
    //   // })
    //  this.willFocusSubscription = this.props.navigation.addListener(
    //   'willFocus',
    //   () => {
    //     console.log("RELOADING DATA");
    //     this.getData();
    //   }
    // );

  }



  async componentDidUpdate() {
    console.log("COMPONENT DID UPDATE")
    console.log("time_gas de this.props.route")
    console.log(this.props.route.params.datosGasolinera.time_gas)
    console.log("time_gas de this.state")
    console.log(this.state.datosGasolinera.time_gas)
    if (this.props.route.params.datosGasolinera !== undefined) {
      if (this.state.datosGasolinera != this.props.route.params.datosGasolinera) {
        console.log("UPDATEado")
        this.setState({ datosGasolinera: this.props.route.params.datosGasolinera })
        this.setState({ 
          horario: this.prettySchedule(this.props.route.params.datosGasolinera.time_gas, FORMAT_EDIT) 
        })
      }
    }
  }

  async getData() {
    //Poner un Toast Loading mientras se realiza esta llamada para informar al usuario que se está cargando
    //Llamar a la función del back para ver los datos de la gasolinera que se quiere
    console.log("Voy a llamar a la funcion del back para obtener los datos de la gasolinera a partir del ID")
    await getGasStationsById(this.props.route.params.idGasolinera).then(data => {
      this.setState({ datosGasolinera: data })
      console.log(this.state.datosGasolinera)
    })
    await allServices().then(data => {
      console.log("All services")
      // console.log(data)
      this.setState({ allServices: data })
    })
    this.setState({
      horario: this.prettySchedule(JSON.stringify(this.state.datosGasolinera.time_gas), FORMAT_BACK)
    })
  }


  //No llamar aqui a BD ya qe al ser async, volveria a llamar a render, que volveria a llamar a componentDidUpdate, generarando infinitas callbacks
  // componentDidUpdate(prevProps) {
  //   console.log("COMPONENT DID UPDATE")

  // }

  //Re-render component when data changes
  // componentWillMount() {
  //   console.log("COMPONENTE WILL MOUNT")

  //   DeviceEventEmitter.addListener('Listener data', async (e)=>{
  //     //reload data
  //     console.log("RELOADING DATA");
  //     this.getData();

  //     })
  // }



  //Muestra en filas separando por '\\n'
  processTime() {
    let timeGas = this.state.datosGasolinera.time_gas;

    if (timeGas !== undefined) {
      let arrayParts: Array<String> = timeGas.split('\\n');
      return (
        <View style={{ backgroundColor: 'white' }}>
          {
            arrayParts.map((parte, index) => (
              <Text style={styles.plain}>{parte}</Text>
            ))
          }
        </View>
      )
    }
  }

  showServices = () => {
    var a = ['Hola1', 'hola2']
    let listServices = this.state.datosGasolinera.services_gas;
    if (listServices !== undefined) {
      let services = JSON.parse(servicesListToIcon);
      return (
        listServices.map((service_name, l) => (
          <View style={styles.containerServices}>
            <Text style={styles.plain}>{services.service_name}</Text>
            <Icon
              reverse
              style={styles.servicesIcon}
              name={services[service_name]}
              type='font-awesome'
              color='black'
              size={15}
              onPress={() => Alert.alert('Service:' + service_name)}
            />
          </View>
        )
        )
      )
    } else {
      return (
        <Text style={{ color: 'black' }}>Indefinido</Text>
      )
    }
  }

  //Print table with gas and price data this.state.datosGasolinera.fuels_gas
  processTableAndPrint = () => {
    if (this.state.datosGasolinera.fuels_gas) {

      const columnsFuel = [
        {
          title: 'Type',
          dataIndex: 'type',
          width: 105
        },
        {
          title: 'Price/l',
          dataIndex: 'price',
          width: 140
        },
      ];

      let dataTable = [];
      const arrayFuels = this.state.datosGasolinera.fuels_gas;
      for (let i = 0; i < arrayFuels.length; i++) {
        let objectPush = {
          'type': arrayFuels.[i].id_fuel,
          'price': arrayFuels.[i].price_fuel
        }
        dataTable.push(objectPush);
      }
      return (
        <Table height={100} columnWidth={60} columns={columnsFuel} dataSource={dataTable} />
      )
    } else {
      return (
        <Text style={{ color: 'black' }}>Indefinido</Text>
      )
    }
  }

  //Return array of last fuel prices
  getLastFuelPrices(fuelType) {
    //datos del fuel fuelType
    if (this.state.datosGasolinera !== undefined) {
      console.log("FUELS_GAS");
      console.log(this.state.datosGasolinera.fuels_gas);
      let data = this.state.datosGasolinera.fuels_gas.filter(index => {
        // console.log("INDEX");
        // console.log(index);
            return (index.id_fuel==fuelType)
      }
      );
      
        console.log("getFUEL: ");
        console.log(data[0].last_prices);
        return data[0].last_prices;
    }
  }

  //Prints graphic with last fuel prices
  printGraphic = (fuelType) => {
    //Contantes para grafica, updatear al leer de bd
    const dataX = [1, 2, 3, 4, 5];
    if (fuelType != "") {
      // const data = [1.10, 1.12, 1.09, 1.05, 1.0];
      console.log("FUEL TYPE")
      console.log(fuelType)
      const data = this.getLastFuelPrices(fuelType);
      const contentInset = { top: 20, bottom: 20 }
      return (
        <View style={styles.pricesView}>
          <YAxis
            data={data}
            contentInset={contentInset}
            svg={{
              fill: 'grey',
              fontSize: 10,
            }}
            numberOfTicks={15}
            formatLabel={(value) => `${value}€`}
          />
          <XAxis
            style={{ marginHorizontal: -10 }}
            data={[50, 10, 40, 95, -4, -24, 85, 91, 35, 53, -53, 24, 50, -20, -80]}
            formatLabel={(value, index) => `Day ${index}`}
            contentInset={{ left: 10, right: 10 }}
            svg={{ fontSize: 10, fill: 'black' }}

          />
          <LineChart
            style={{ flex: 1, marginLeft: 16 }}
            data={data}
            svg={{ stroke: 'rgb(134, 65, 244)' }}
            contentInset={contentInset}
          >
            <Grid />
          </LineChart>
        </View>
      )
    } else {

    }
  }

  setFuelType = (fuelType) => {
    this.setState({
      fuelType: fuelType
    })
  }

  changeUpdateModalState = () => {
    this.setState({
      modalUpdateVisible: !this.state.modalUpdateVisible
    })
  }
  setFuel = (fuelType) => {
    this.setState({
      fuel: fuelType
    })
  }

  setPrice = (price) => {
    this.setState({
      price: price
    })
  }

  updatePrice = async () => {
    const datos = {
      fuel: this.state.fuel,
      id_gas: this.props.route.params.idGasolinera,
      price: this.state.price.replace(",", ".")
    }
    await updatePriceMan(datos).then(data => {
      if (data == "Changed correctly") {
        Alert.alert("Success!", "The price has been updated correctly!", [{ text: "OK", onPress: () => this.changeUpdateModalState() }]);
      }
      else if (data == "Fuel added!") {
        Alert.alert("Success!", "You succesfully added a new type of fuel!", [{ text: "OK", onPress: () => this.changeUpdateModalState() }]);
      }
      else {
        Alert.alert("ERROR: ", "General ERROR: The price was not updated. Please try again later.", [{ text: "OK", onPress: () => this.changeUpdateModalState() }])
      }
    })
  }

  printIconEdit = (nameScreen) => {
    if (nameScreen == "editPrice") {
      return (
        <Icon
          reverse
          // style={styles.servicesIcon}
          name="pencil-square"
          type='font-awesome'
          color='blue'
          size={15}
          onPress={() => this.changeUpdateModalState()}

        />
      )
    }
    else {
      return (
        <Icon
          reverse
          // style={styles.servicesIcon}
          name="pencil-square"
          type='font-awesome'
          color='blue'
          size={15}
          onPress={() => this.props.navigation.navigate(nameScreen, { datosGasolinera: this.state.datosGasolinera, allServices: this.state.allServices })}

        />
      )
    }
  }

  prettySchedule = (schedule: string, format: string) => {
    if (format == FORMAT_BACK){
      return( schedule
      .replace(/\\\\n/g, "\n").replace(/\"/g, "").replace(/ /g, "\t")
      .replace(/Fri:/g, "Fri:  ").replace(/-/g, "\t-\t")
      .replace(/Sat:/g, "Sat: "))
    } else if (format == FORMAT_EDIT) {
      return( schedule
      .replace(/\\n/g, "\n").replace(/\"/g, "").replace(/ /g, "\t")
      .replace(/Fri:/g, "Fri:  ").replace(/-/g, "\t-\t")
      .replace(/Sat:/g, "Sat: "))
    } else {
      return "FORMAT ERROR"
    }
  } 

  render() {
    return (
      <ScrollView>
        <View style={styles.container}>
          <View style={styles.containerTop}>
            <Text style={styles.mainTitle}>Your Gas Station is:</Text>
            <Text style={styles.plainBold}>
              <Icon
                name='location-on'
                type='material'
                color='black'
                size={30}
                onPress={() => console.log('hello')}
              />
              {this.state.datosGasolinera.street_gas}
            </Text>
            {/* Horizontal line */}
            <View style={styles.separator}>

            </View>
          </View>
          <View style={styles.containerServices}>
            <Text style={styles.plainBold}>
              {'Services: '}
            </Text>
            <ScrollView horizontal={true} style={styles.servicesView}>
              {/* Prueba a llamar a los iconos dinamicamente */}
              {this.showServices()}
            </ScrollView >
            {this.printIconEdit("ManagerEditServices")}

          </View>
          <View style={styles.containerSchedule}>
            <Text style={styles.plainBold}>
              {'Schedule:'}
            </Text>
            <View style={styles.scheduleView}>
              <Text style={styles.plain}>
                {console.log("------------------")}
                {console.log(this.state.horario)}
                {console.log("------------------")}
                {this.state.horario}
              </Text>
            </View>
            {this.printIconEdit("ManagerEditSchedule")}
          </View>
          <View style={styles.containerFuel}>
            <Text style={styles.plainBold}>
              {'Available fuel:'}
            </Text>
            <View style={styles.fuelView}>
              {this.processTableAndPrint()}
              {this.printIconEdit("editPrice")}
            </View>
          </View>
          <View style={styles.containerPrices}>
            <Text style={styles.plainBold}>
              {'Latest prices:'}
            </Text>
            <Picker
              selectedValue={this.state.fuelType}
              style={{ height: 50, width: 200 }}
              onValueChange={(itemValue, itemIndex) => this.setFuelType(itemValue)}
            >
              <Picker.Item label="Gas" value="Gas" />
              <Picker.Item label="Gas Premium" value="Gas Premium" />
              <Picker.Item label="Diesel" value="Diesel" />
              <Picker.Item label="Diesel Premium" value="Diesel Premium" />
            </Picker>
            {this.printGraphic(this.state.fuelType)}
          </View>
        </View>
        <Modal
          animationType="slide"
          transparent={true}
          visible={this.state.modalUpdateVisible}
          onRequestClose={() => {
            Alert.alert("Popup has been closed.");
          }}>
          <View style={styles.containerModal}>
            <View style={styles.modalView}>
              <Text style={styles.modalText}>Select the fuel and type the price</Text>
              <Picker
                selectedValue={this.state.fuel}
                style={{ height: 200, width: 200 }}
                onValueChange={(itemValue, itemIndex) => this.setFuel(itemValue)}
              >
                <Picker.Item label="Gas" value="Gas" />
                <Picker.Item label="Gas Premium" value="Gas Premium" />
                <Picker.Item label="Diesel" value="Diesel" />
                <Picker.Item label="Diesel Premium" value="Diesel Premium" />
              </Picker>
              <TextInput
                style={{ height: 40, width: 200, borderColor: 'gray', borderWidth: 1, textAlign: 'center', fontSize: 22 }}
                keyboardType={'numeric'}
                onChangeText={text => this.setPrice(text)}
                value={this.state.price}
              />
              <View style={styles.containerButtons}>
                <Button
                  containerStyle={styles.button}
                  title="Save"
                  onPress={() => {
                    this.updatePrice();
                  }}
                />
                <Text>{"\t"}</Text>
                <Button
                  containerStyle={styles.button}
                  title="Cancel"
                  onPress={() => {
                    this.changeUpdateModalState();
                  }}
                />
              </View>
            </View>
          </View>
        </Modal>
      </ScrollView>
    );
  }


}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white'
  },
  containerTop: {
    alignItems: 'center',
    paddingBottom: '5%',
    backgroundColor: 'white'
  },
  containerServices: {
    backgroundColor: 'white',
    flexDirection: 'row'
  },
  containerSchedule: {
    backgroundColor: 'white',
    flexDirection: 'row'
  },
  containerFuel: {
    backgroundColor: 'white',
    // flexDirection: 'row'
  },
  containerPrices: {
    backgroundColor: 'white',
    // flexDirection: 'row'
  },
  containerButtons: {
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'center'
  },
  containerModal: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: 'transparent',
    marginTop: 22
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5
  },
  modalText: {
    color: 'black'
  },
  button: {
    backgroundColor: "yellow",
    alignSelf: "center",
    width: "40%",
    borderRadius: 5,
    marginTop: 10,
    marginBottom: 50
  },
  mainTitle: {
    fontSize: 25,
    fontWeight: 'bold',
    paddingTop: '10%',
    // paddingBottom: '10%',
    color: 'black',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'black'
  },
  plain: {
    fontSize: 17,
    color: 'black',
  },
  plainBold: {
    fontSize: 23,
    color: 'black',
    fontWeight: "900",
    paddingBottom: '5%',
    paddingTop: '5%',
    paddingLeft: '5%'
  },
  servicesIcon: {

  },
  servicesView: {
    paddingTop: '2.5%',
    backgroundColor: 'white',
    flexDirection: 'row',
  },
  scheduleView: {
    paddingTop: '5%',
    paddingLeft: '2.5%',
    backgroundColor: 'white',
    flexDirection: 'row'
  },
  fuelView: {
    backgroundColor: 'white',
    alignItems: 'center'
  },
  pricesView: {
    height: 300,
    width: '100%',
    backgroundColor: 'white',
    flexDirection: 'row'
  },
  separator: {
    marginBottom: 5,
    height: 1,
    width: '100%',
  },
  //table

});
