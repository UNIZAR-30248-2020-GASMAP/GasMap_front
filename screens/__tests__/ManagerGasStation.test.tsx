import React from 'react';
import renderer from 'react-test-renderer';

import ManagerGasStation, { FORMAT_BACK, FORMAT_EDIT } from '../ManagerGasStation';

test('Correct Rendered', () => {
    const tree = renderer.create(<ManagerGasStation />).toJSON();
    expect(tree).toMatchSnapshot()
});

test('prettySchedule BACK', () => {
    const mng = new ManagerGasStation(null)
    const schedule = "Mon: 08:00-23:00\\nTue: 07:00-23:00\\nWed: 08:00-23:00\\nThu: 09:00-23:00\\nFri: 10:00-23:00\\nSat: 06:00-00:00\\nSun: 06:00-00:00\\n"
    const prettyScheduleString = "Mon:	08:00	-	23:00\\nTue:	07:00	-	23:00\\nWed:	08:00	-	23:00\\nThu:	09:00	-	23:00\\nFri:  	10:00	-	23:00\\nSat: 	06:00	-	00:00\\nSun:	06:00	-	00:00\\n"
    
    expect(mng.prettySchedule(schedule,FORMAT_BACK)).toEqual(prettyScheduleString)
});

test('prettySchedule EDIT', () => {
    const mng = new ManagerGasStation(null)
    const schedule = "Mon: 08:00-23:00\\nTue: 07:00-23:00\\nWed: 08:00-23:00\\nThu: 09:00-23:00\\nFri: 10:00-23:00\\nSat: 06:00-00:00\\nSun: 06:00-00:00\\n"
    const prettyScheduleString = "Mon:	08:00	-	23:00\nTue:	07:00	-	23:00\nWed:	08:00	-	23:00\nThu:	09:00	-	23:00\nFri:  	10:00	-	23:00\nSat: 	06:00	-	00:00\nSun:	06:00	-	00:00\n"
    
    expect(mng.prettySchedule(schedule,FORMAT_EDIT)).toEqual(prettyScheduleString)
});
