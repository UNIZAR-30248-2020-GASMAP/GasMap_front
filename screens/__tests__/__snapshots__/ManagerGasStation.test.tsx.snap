// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct Rendered 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "paddingBottom": "5%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 25,
                "fontWeight": "bold",
                "paddingTop": "10%",
              },
            ]
          }
        >
          Your Gas Station is:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "height": 1,
                "marginBottom": 5,
                "width": "100%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "backgroundColor": "white",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          Services: 
        </Text>
        <RCTScrollView
          horizontal={true}
          style={
            Object {
              "backgroundColor": "white",
              "flexDirection": "row",
              "paddingTop": "2.5%",
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                  },
                  Object {
                    "color": "black",
                  },
                ]
              }
            >
              Indefinido
            </Text>
          </View>
        </RCTScrollView>
        <View
          style={
            Object {
              "borderRadius": 19,
              "height": 34,
              "margin": 7,
              "overflow": "hidden",
              "width": 34,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "borderRadius": 19,
                  "height": 34,
                  "justifyContent": "center",
                  "width": 34,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "backgroundColor": "white",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          Schedule:
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "backgroundColor": "white",
                "flexDirection": "row",
                "paddingLeft": "2.5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "black",
                  "fontSize": 17,
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "borderRadius": 19,
              "height": 34,
              "margin": 7,
              "overflow": "hidden",
              "width": 34,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "blue",
                  "borderRadius": 19,
                  "height": 34,
                  "justifyContent": "center",
                  "width": 34,
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "backgroundColor": "white",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          Available fuel:
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Indefinido
          </Text>
          <View
            style={
              Object {
                "borderRadius": 19,
                "height": 34,
                "margin": 7,
                "overflow": "hidden",
                "width": 34,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "blue",
                    "borderRadius": 19,
                    "height": 34,
                    "justifyContent": "center",
                    "width": 34,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "backgroundColor": "white",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "black",
                "fontSize": 23,
                "fontWeight": "900",
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingTop": "5%",
              },
            ]
          }
        >
          Latest prices:
        </Text>
        <View
          style={
            Object {
              "height": 50,
              "width": 200,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Gas",
                  "textColor": undefined,
                  "value": "Gas",
                },
                Object {
                  "label": "Gas Premium",
                  "textColor": undefined,
                  "value": "Gas Premium",
                },
                Object {
                  "label": "Diesel",
                  "textColor": undefined,
                  "value": "Diesel",
                },
                Object {
                  "label": "Diesel Premium",
                  "textColor": undefined,
                  "value": "Diesel Premium",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 22,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderRadius": 20,
                "elevation": 5,
                "margin": 20,
                "padding": 35,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Select the fuel and type the price
          </Text>
          <View
            style={
              Object {
                "height": 200,
                "width": 200,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Gas",
                    "textColor": undefined,
                    "value": "Gas",
                  },
                  Object {
                    "label": "Gas Premium",
                    "textColor": undefined,
                    "value": "Gas Premium",
                  },
                  Object {
                    "label": "Diesel",
                    "textColor": undefined,
                    "value": "Diesel",
                  },
                  Object {
                    "label": "Diesel Premium",
                    "textColor": undefined,
                    "value": "Diesel Premium",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "gray",
                "borderWidth": 1,
                "fontSize": 22,
                "height": 40,
                "textAlign": "center",
                "width": 200,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                },
                Object {
                  "backgroundColor": "white",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "yellow",
                  "borderRadius": 5,
                  "marginBottom": 50,
                  "marginTop": 10,
                  "overflow": "hidden",
                  "width": "40%",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Save
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                  },
                  undefined,
                ]
              }
            >
              	
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "yellow",
                  "borderRadius": 5,
                  "marginBottom": 50,
                  "marginTop": 10,
                  "overflow": "hidden",
                  "width": "40%",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
